### PR TITLE
Add connector usage to the testbed

### PR DESCRIPTION
**Pull request by humivo**
**Description:** Added a new component to the testbed called DataConnectors which allows connectors to be added to the testbed config. Also, added a sample connector correctness test using the routingconnector as an example of the usage.

**Link to tracking Issue:** #30165 

**Testing:** Sample correctness test using routingconnector.

**Documentation:** Will update the testbed README with new addition.

### CODE REVIEW

**Review by mx-psi — APPROVED** [review]
cc @djaglowski in case you want to review

File: testbed/dataconnectors/routing.go [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+	return &RoutingDataConnector{DataConnectorBase: testbed.DataConnectorBase{ReceiverDataType: receiverDataType}}
+}
+
+func (rc *RoutingDataConnector) GenConfigYAMLStr() string {
+	// Note that this generates an exporter config for agent.
+	return `
+  routing:
+    table:
+      - statement: route()
+        pipelines: [traces/out]`
```
Comment (djaglowski) [inline testbed/dataconnectors/routing.go:26]: Do you plan to iterate on this in a future PR? I get that you're mostly just adding the framework now but this particular test doesn't appear to validate much.

**Review by djaglowski — APPROVED** [review]
Awesome, thanks @humivo

File: testbed/dataconnectors/routing.go (hunk 1 above)
Comment (humivo) [reply testbed/dataconnectors/routing.go:26]: Yes we could iterate on this in a future PR but just wanted to add this test to validate that connectors are included within the config and the connector is working as expected.

**Review by djaglowski — APPROVED** [review]

**Review by bryan-aguilar — APPROVED** [review]

**Pull request merged**